### ISSUE TITLE
Azure Blob Store support

### ISSUE THREAD

**Issue by dmytrohridin** [tusdotnet/tusdotnet#90]
Hello,
Is there are any plans for adding Azure blob storage as TusStore?
Thank you

**Comment by smatsson** [comment]
Hi,

There are no current plans to have this out of the box in tusdotnet as we do not want vendor lock in. 

Block and Page blob storage does not really share the same model as the tus protocol as there is no way of updating these types of blobs without downloading the existing one, modifying it and uploading it again. [TusAzureStore](https://github.com/smatsson/TusAzureStore) is a tusdotnet store POC using Append blobs. Last time I used Append blobs I found them very slow compared to direct disk access.

The alternatives that you have are basically:
* Use Azure Files instead which is supported by mounting the Azure Files account as a local drive (https://docs.microsoft.com/sv-se/azure/storage/files/storage-files-introduction)
* Write [your own store](https://github.com/tusdotnet/tusdotnet/wiki/Custom-data-store) and use the TusAzureStore POC as base.
